protocol/contracts/beanstalk/init/reseed/L2/ReseedBean.sol

```diff
@@ -170,68 +184,157 @@ contract ReseedBean {
     }
 
     function deployUpgradebleWell(
-        address implementation,
+        IERC20[] memory tokens,
+        Call memory wellFunction,
+        Call[] memory pumps,
         bytes32 salt,
         string memory name,
         string memory symbol
     ) internal {
-        address well = address(
+        
+        // Encode well data
+        (bytes memory immutableData, bytes memory initData) =
+            encodeWellDeploymentData(AQUIFER, tokens, wellFunction, pumps);
+
+        // Bore upgradeable well
+        address _well = IAquifer(AQUIFER).boreWell(
+            UPGRADEABLE_WELL_IMPLEMENTATION, immutableData, initData, salt
+        );
+
+        // Deploy proxy
+        address wellProxy = address(
             new ERC1967Proxy{salt: salt}(
-                implementation,
+                _well,
                 abi.encodeCall(IWellUpgradeable.init, (name, symbol))
             )
         );
-        console.log(name);
-        console.log("well proxy deployed at:", well);
     }
 
-    function deployUpgradableWells() internal {
+    function deployUpgradableWells(address bean) internal {
+        // tokens
+        IERC20[] memory tokens = new IERC20[](2);
+        tokens[0] = IERC20(bean);
+        tokens[1] = IERC20(WETH);
+
+        // cp2 
+        // TODO: change data if needed
+        Call memory cp2 = Call(CONSTANT_PRODUCT_2, abi.encode("beanWF"));
+
+        // stable2
+        // TODO: change data if needed
+        Call memory stable2 = Call(STABLE_2, abi.encode("beanStable"));
```

Comment:
the stable2 implementation requires the decimals of the token to be encoded here (where the index matches the token index - see stable2 implementation). Could you fix this so that this differs on which stable is being bored?

Reply:
Did some research and looks like both usdc and usdt have 6 decimals in all chains so i just encoded that along with the bean decimals here https://github.com/BeanstalkFarms/Beanstalk/pull/1012/commits/d41b0275bbe22b8520d04ee5dfe9e51ba3b4af21